### DISCUSSION TITLE
Integrating Multiple Phase Profiles in RIS Modeling for Indoor Scenarios

### DISCUSSION THREAD

**Discussion by ekilcioglu** [NVlabs/sionna#583] [Q&A] [answered]
Hello,

I have a question regarding one of the properties of RIS modeling in the ray-tracing tool. In an indoor office scenario, I want the RIS to steer its beams towards two different positions. To achieve this, I first set the "num_modes" property of the RIS object to 2 and used the phase_gradient approach. After defining, two different phase profiles are defined by the phase gradient helper function for two different positions.

My question is: how do you integrate these two different phase profiles into the same RIS when calculating the coverage map? Do you unify these phase profiles into a single phase profile at the end?

I am asking this because in my approach without using any num_modes property of the RIS object, I was considering dividing the RIS into two halves. By using the phase gradient approach in the literature paper, I computed the related phase gradient values for both positions and I defined a single phase profile where half of the 2D phase profile matrix is for the first position and the other half is for the other position so that I unify the phase profile matrices to obtain a single phase profile matrix for the RIS. At the end, I obtained similar results with "num_modes" usage approach in terms of the coverage map and RIS coverage gain.

However, I am curious whether it’s possible to implement your approach. This is important because, by the end of our project, I need to deliver a single phase profile function given the incidence and desired reflection directions, which can cover multiple blind spots in a given indoor scenario so that it will be manufactured according to that.

Anyone has an idea? Thank you so much for your help!

Best regards

**Comment by jhoydis** [comment, marked as answer]
Hi @ekilcioglu,

What you actually configure for the RIS is the spatial modulation coefficient as described in [Eq. (43)](https://nvlabs.github.io/sionna/em_primer.html#equation-spatial-modulation-coefficient).
Even with multiple reradiation modes, your RIS has a single spatial modulation coefficient for each unit cell, which is simply the superposition of the coefficients of all reradiation modes.

There is no need to split the RIS into multiple parts, etc.

Hope that helps.

**Comment by ekilcioglu** [comment]
Yes, thank you for the clarification!

Best regards
